src/libs/OptionsListUtils.js

```diff
@@ -654,28 +664,28 @@ function getSidebarOptions(
 }
 
 /**
- * Helper method that returns the text to be used for the header's message and title (if any)
- *
+ * Helper method that returns the text to be used for the header's message and title (if any), the isValid
+ * property decides if we should show the search results for the given search value.
  * @param {Boolean} hasSelectableOptions
  * @param {Boolean} hasUserToInvite
  * @param {String} searchValue
  * @param {Boolean} [maxParticipantsReached]
- * @return {String}
+ * @return {Object}
  */
 function getHeaderMessage(hasSelectableOptions, hasUserToInvite, searchValue, maxParticipantsReached = false) {
     if (maxParticipantsReached) {
-        return translate(preferredLocale, 'messages.maxParticipantsReached');
+        return {message: translate(preferredLocale, 'messages.maxParticipantsReached'), isValid: true};
     }
 
-    if (!hasSelectableOptions && !hasUserToInvite) {
-        if (/^\d+$/.test(searchValue)) {
-            return translate(preferredLocale, 'messages.noPhoneNumber');
-        }
+    if (isSearchingInvalidPhone(searchValue)) {
+        return {message: translate(preferredLocale, 'messages.noPhoneNumber'), isValid: false};
+    }
 
-        return searchValue;
+    if (!hasSelectableOptions && !hasUserToInvite) {
+        return {message: searchValue, isValid: true};
     }
 
-    return '';
+    return {message: '', isValid: true};
```

Comment:
We should probably decouple the header `message` from the rendering of the `SectionList`.  If needed, we can just add another prop that is along the lines of shouldShowSectionList